src/drivers/SpiNorFlash.h

```diff
@@ -1,60 +1,116 @@
 #pragma once
+
 #include <cstddef>
 #include <cstdint>
+#include <concepts>
 
 namespace Pinetime {
   namespace Drivers {
-    class Spi;
-    class SpiNorFlash {
-    public:
-      explicit SpiNorFlash(Spi& spi);
-      SpiNorFlash(const SpiNorFlash&) = delete;
-      SpiNorFlash& operator=(const SpiNorFlash&) = delete;
-      SpiNorFlash(SpiNorFlash&&) = delete;
-      SpiNorFlash& operator=(SpiNorFlash&&) = delete;
-
-      struct __attribute__((packed)) Identification {
-        uint8_t manufacturer = 0;
-        uint8_t type = 0;
-        uint8_t density = 0;
-      };
+    template <typename T>
```

Comment:
suggest using more specific name for the type

```suggestion
    template <typename MemoryImpl>
```